corpus/pages/views.py

```diff
@@ -26,9 +28,27 @@ def about_us(request):
         },
     )
 
+
 def impulse(request):
 
     return render(
         request,
         "pages/impulse.html",
-    )
\ No newline at end of file
+    )
+
+
+def sig(request, sig_name):
+    sig_data = get_object_or_404(SIG, name=sig_name)
+
+    # Retrieve the related society details using the SIG instance
+    society_linked_to_sig = Society.objects.filter(sig=sig_data.id)
```

Comment:
If you make this into a relationship field in the model, you might be able to use something like `sig_data.societies` or something. Check what exactly works.

Reply:
done